xds/src/main/java/io/grpc/xds/internal/security/certprovider/CertProviderSslContextProvider.java

```diff
@@ -149,22 +150,40 @@ public final void updateTrustedRoots(List<X509Certificate> trustedRoots) {
     updateSslContextWhenReady();
   }
 
+  @Override
+  public final void updateSpiffeRoots(Map<String, List<X509Certificate>> spiffeRoots) {
+    savedSpiffeRoots = spiffeRoots;
+    updateSslContextWhenReady();
+  }
+
   private void updateSslContextWhenReady() {
     if (isMtls()) {
       if (savedKey != null && savedTrustedRoots != null) {
         updateSslContext();
         clearKeysAndCerts();
       }
+      if (savedSpiffeRoots != null) {
+        updateSslContext();
+        clearKeysAndCerts();
+      }
     } else if (isClientSideTls()) {
       if (savedTrustedRoots != null) {
         updateSslContext();
         clearKeysAndCerts();
       }
+      if (savedSpiffeRoots != null) {
+        updateSslContext();
+        clearKeysAndCerts();
+      }
     } else if (isServerSideTls()) {
       if (savedKey != null) {
         updateSslContext();
         clearKeysAndCerts();
       }
+      if (savedSpiffeRoots != null) {
```

Comment:
I think this can be skipped here: there is no need to use the SPIFFE trust map for server-side TLS.

Reply:
Friendly ping, it looks like this comment was not addressed?